pkg/miner/miner.go

```diff
@@ -238,9 +238,15 @@ func (a *MicroblockMiner) mineMicro(ctx context.Context, rest restLimits, blockA
 	}
 
 	priv := keyPair.Secret
+	if newBlock.Version >= proto.ProtoBlockVersion {
+		err = newBlock.SetTransactionsRoot(a.scheme)
```

Comment:
Can we move `SetTransactionRoot()` call to `CreateBlock()` function?
It is used in multiple places, so might be better this way.